spec/system/post_spec.rb

```diff
@@ -0,0 +1,119 @@
+    @comment3 = Comment.create(author: @tom, post: @post1, text: 'This is my first comment @post1')
+    @comment4 = Comment.create(author: @jerry, post: @post1, text: 'Hey , tom')
+  end
+
+  describe 'User visits the post index page' do
+    scenario 'I can see the user profile picture' do
+      visit user_posts_path(@tom)
+
+      assert_selector "img[src='#{@tom['photo']}']"
+    end
+
+    scenario 'I can see the user username' do
+      visit user_posts_path(@tom)
+
+      assert_text @tom.name
+    end
+
+    scenario 'I can see the number of posts the user has written' do
+      visit user_posts_path(@tom)
+
+      assert_text "Number of posts: #{@tom.posts_counter}"
+    end
+
+    scenario 'I can see a post title' do
+      visit user_posts_path(@tom)
+
+      assert_text @post4.title
+    end
+
+    scenario 'I can see some of the post body' do
+      visit user_posts_path(@tom)
+
+      assert_text @post3.text
+    end
+
+    scenario 'I can see the first comments on a post' do
+      visit user_posts_path(@tom)
+
+      assert_text @post3.comments.first.text
+    end
+
+    scenario 'I can see how many comments a post has' do
+      visit user_posts_path(@tom)
+
+      assert_text "Comments: #{@post3.comments_counter}"
+    end
+
+    scenario 'I can see a section for pagination if there are more posts than fit on the view' do
+      visit user_posts_path(@tom)
+
+      assert_text '2 Next'
+    end
+
+    scenario 'When I click on a post, it redirects me to that post show page' do
+      visit user_posts_path(@tom.id)
+      within("#post-card-#{@post3.id}") do
+        click_on @post3.title
+      end
+      expect(page).to have_current_path(user_post_path(@tom.id, @post3.id))
+    end
+  end
+
+  describe 'User visits the post show page' do
+    scenario 'I can see a post title' do
+      visit user_post_path(@tom, @tom.posts.first)
+
+      assert_text @tom.posts.first.title
+    end
+
+    scenario 'I can see who wrote the post' do
+      visit user_post_path(@tom, @tom.posts.first)
+
+      assert_text "#{@tom.posts.first.title} by #{@tom.name}"
+    end
+
+    scenario 'I can see how many comments a post has' do
+      visit user_post_path(@tom, @tom.posts.first)
+
+      assert_text "Comments: #{@tom.posts.first.comments_counter}"
+    end
+
+    scenario 'I can see how many likes a post has' do
+      visit user_post_path(@tom, @tom.posts.first)
+
+      assert_text "#{@tom.posts.first.likes_counter} Likes"
+    end
+
+    scenario 'I can see the post body' do
+      visit user_post_path(@tom, @tom.posts.first)
+
+      assert_text @tom.posts.first.text
+    end
+
+    scenario 'I can see the username of each commentor and the comment each commentor left.' do
+      visit user_post_path(@tom, @tom.posts.first)
+
+      assert_text "#{@tom.posts.first.comments.first.author.name}:#{@tom.posts.first.comments.first.text}"
+    end
+  end
+end
```

Comment:
- [ ] Based on the requirements, each page should have a test file that contains all tests written for that page. This means that you should have separate test files for the `Post#Index` page and the `Post#Show` page and write the tests in them based on the requirement for that page. 👍